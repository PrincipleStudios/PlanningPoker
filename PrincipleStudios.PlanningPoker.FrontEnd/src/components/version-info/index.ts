export { VersionInfoContainer as VersionInfo } from './version-info.container';
