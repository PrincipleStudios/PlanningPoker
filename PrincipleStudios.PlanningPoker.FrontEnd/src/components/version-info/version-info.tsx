import { EnvironmentInfoResponse } from 'src/api/models/EnvironmentInfoResponse';

export function VersionInfo({ environmentInfo }: { environmentInfo: EnvironmentInfoResponse }) {
	return (
		<>
			<dl>
				<dt>Git Commit Hash:</dt>
				<dd>{environmentInfo.gitHash}</dd>
			</dl>
		</>
	);
}
