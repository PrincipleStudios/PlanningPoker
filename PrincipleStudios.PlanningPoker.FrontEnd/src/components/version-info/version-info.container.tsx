import { map } from 'rxjs/operators';
import { useApi } from 'src/core/utils/api';
import { useObservable } from 'src/core/utils/use-observable';
import { VersionInfo } from './version-info';

export function VersionInfoContainer() {
	const api = useApi();
	const infoApiCall = api.getInfo();

	const value = useObservable(
		() => infoApiCall.pipe(map((response) => (response.statusCode === 200 ? response.data : {}))),
		{ gitHash: 'local' }
	);

	if (value == null) return null;

	return <VersionInfo environmentInfo={value} />;
}
