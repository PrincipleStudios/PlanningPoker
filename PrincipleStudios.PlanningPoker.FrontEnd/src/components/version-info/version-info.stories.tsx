import { ComponentStory, ComponentMeta } from '@storybook/react';
import { VersionInfo } from './version-info';

export default {
	title: 'Components/VersionInfo',
	component: VersionInfo,
} as ComponentMeta<typeof VersionInfo>;

const Template: ComponentStory<typeof VersionInfo> = (args) => (
    <VersionInfo {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
    gitHash: 'storybook-githash'
};
