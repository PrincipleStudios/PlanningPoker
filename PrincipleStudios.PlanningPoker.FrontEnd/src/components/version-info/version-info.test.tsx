import { render } from '@testing-library/react';
import { VersionInfo } from './version-info';

describe('version-info component', () => {
	it('renders the provided gitHash version', () => {
		const { baseElement } = render(<VersionInfo gitHash="some-git-hash" />);
		expect(baseElement).toMatchSnapshot();
	});
});
