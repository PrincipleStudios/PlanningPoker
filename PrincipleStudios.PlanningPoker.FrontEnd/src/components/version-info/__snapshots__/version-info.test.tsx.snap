// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`version-info component renders the provided gitHash version 1`] = `
<body>
  <div>
    <dl
      class="grid grid-cols-2"
    >
      <dt
        class="font-bold"
      >
        Git Commit Hash:
      </dt>
      <dd>
        some-git-hash
      </dd>
    </dl>
  </div>
</body>
`;
